.github/workflows/bump-dependencies.yml

```diff
@@ -0,0 +1,179 @@
+name: Bump dependencies
+
+on:
+  # For PR creation:
+  workflow_dispatch:
+  schedule:
+    - cron: '13 8 * * 6'  # once a week on Saturdays at 08:13
+
+  # For branch deletion:
+  # We have to use pull_request_target because pull_request does not trigger on closed PRs with merge conflicts:
+  # https://github.com/orgs/community/discussions/26304
+  # WARNING: pull_request_target is dangerous as the run can be influenced by the PR origin.
+  pull_request_target:
+    types:
+      - closed
+
+  # For rebasing:
+  push:
+    branches: [master]
+
+permissions:
+  pull-requests: write
+  # The main repository writes are done using a deploy key as we can't grant
+  # workflows: permission here, which is required when pushing to .github/workflows.
+  # We need basic access for branch deletion anyway:
+  contents: write
+
+jobs:
+  create-prs:
+    name: Update ${{ matrix.components.name }} if necessary
```

Comment:
These need to be clearly from an automated source.  Flagging that in the title would be good.